velox/exec/tests/TableWriteTest.cpp

```diff
@@ -40,12 +40,40 @@ class TableWriteTest : public HiveConnectorTestBase {
   makeHiveConnectorSplits(
       const std::shared_ptr<TempDirectoryPath>& directoryPath) {
     std::vector<std::shared_ptr<connector::ConnectorSplit>> splits;
-    for (auto& filePath : fs::directory_iterator(directoryPath->path)) {
-      splits.push_back(makeHiveConnectorSplit(filePath.path().string()));
+
+    for (auto& subPath :
+         fs::recursive_directory_iterator(directoryPath->path)) {
+      if (subPath.is_regular_file()) {
+        splits.push_back(makeHiveConnectorSplit(subPath.path().string()));
+      }
     }
+
     return splits;
   }
 
+  uint32_t countFilesRecursively(
```

Comment:
Unit tests still to be revised

Reply:
@gggrace14 Thank you, Ge. Let's makes sure to verify the generated directory names, file content, single and multiple partition keys, single and multiple partitions, a limit on total number of partitions.

Reply:
Test cases are updated to cover generated directory names, file content, single and multiple partition keys, single and multiple partitions. For the limit on the total number of partitions, we have a test case for PartitionIdGenerator, which is where the limit is enforced. Let me know if we also want one test case for TableWriteTest. https://github.com/facebookincubator/velox/blob/2e6f23fb4cf17e6dce61d44b7b3a464c0076c5f0/velox/connectors/hive/tests/PartitionIdGeneratorTest.cpp#L142-L156

Reply:
@gggrace14 It would be nice to have an e2e limit test as well to ensure TableWriter properly fetches the limit from the config and passes it to PartitionIdGenerator.